users/talk.rs

```diff
@@ -0,0 +1,195 @@
+                    break;
+                }
+                termion::event::Key::Ctrl('l') => {
+                    write!(stdout, "{}", termion::clear::All)?;
+                }
+                termion::event::Key::Char('\x07') => {
+                    // Alert character (bell)
+                    write!(stdout, "\x07")?;
+                }
+                termion::event::Key::Char(c) => {
+                    if c == '\n' {
+                        // Send the collected input on Enter
+                        // Format the message with the username and send it
+                        let message = format!("{}: {}\r\n", username, input_buffer);
+                        stream.write_all(message.as_bytes())?;
+                        input_buffer.clear();
+                        writeln!(stdout, "\r")?; // Move to the next line in the terminal
+                    } else {
+                        input_buffer.push(c);
+                        write!(stdout, "{}", c)?; // Echo the character to the terminal
+                    }
+                }
+                termion::event::Key::Backspace => {
+                    if !input_buffer.is_empty() {
+                        input_buffer.pop();
+                        write!(stdout, "\x08 \x08")?;
+                    }
+                }
+                _ => {}
+            }
+            stdout.flush()?;
+        }
+    }
+    Ok(())
+}
+
+/// Handles output received from the connected peer.
+///
+/// This function listens for incoming data from the connected peer, and
+/// writes it to the user's terminal screen. The session continues as long
+/// as the `running` flag is set to `true`.
+///
+/// # Arguments
+///
+/// * `stream` - The TCP stream connected to the peer.
+/// * `running` - An atomic boolean flag indicating whether the session is still active.
+fn handle_output(mut stream: TcpStream, running: Arc<AtomicBool>) -> io::Result<()> {
+    let mut stdout = io::stdout().into_raw_mode()?;
+    let mut buffer = [0; 512];
+    while running.load(Ordering::SeqCst) {
+        let n_read = stream.read(&mut buffer)?;
+        if n_read == 0 {
+            break;
+        }
+        stdout.write_all(&buffer[..n_read])?;
+        stdout.flush()?;
+    }
+    Ok(())
+}
+
+/// Manages the `talk` session by setting up the connection and handling input and output.
+///
+/// This function either starts a listener on the specified address to wait
+/// for incoming connections or connects to a remote address as a client. Once
+/// connected, it spawns separate threads for handling input and output streams.
+///
+/// # Arguments
+///
+/// * `args` - The command-line arguments specifying the address to connect to or listen on.
+fn talk(args: Args) -> Result<(), Box<dyn std::error::Error>> {
+    let running = Arc::new(AtomicBool::new(true));
+    let address = args.address.clone();
+
+    // Inform user about connection
+    println!("Message from <unspecified string>");
```

Comment:
That's not a static text. `<unspecified string>` should instead show who is trying to talk to this tty. Example:
```
Message from TalkDaemon@his_machine...
talk: connection requested by your_name@your_machine.
talk: respond with: talk your_name@your_machine
```
See https://linux.die.net/man/1/talk